### ISSUE TITLE
Buffering state (JPVideoPlayerStatusBuffering) not arriving at the correct time

### ISSUE THREAD

**Issue by om-ha** [newyjp/JPVideoPlayer#204]
Awesome library @newyjp , great updates too squashing those bugs.

I'm using *JPVideoPlayerDelegate* to be notified of the player state. Namely:
```
- (void)playerStatusDidChanged:(JPVideoPlayerStatus)playerStatus;
```

The problem is that the buffering state, *JPVideoPlayerStatusBuffering*, is not notified through the player delegate **exactly/instantly** when the JPVideoPlayer is actually buffering. There's a second or less of delay until the state gets updated and notified through the delegate.

So in other words buffering state, *JPVideoPlayerStatusBuffering*, is being notified **later** than it should be, and when buffering is finished the playing state, *JPVideoPlayerStatusPlaying*, is notified also **later** than it should be.

**Comment by om-ha** [comment]
Could you at least point out where the problem/fix can be please?
Is it *checkBufferingTimer* being 0.3 seconds? in *JPVideoPlayer.m***->***startCheckBufferingTimer*

**Comment by newyjp** [comment]
Yes, it is. we can not receive notification of buffering from `AVPlayer`.

**Comment by xl20071926** [comment]
大佬请教一下我这边也遇到类似问题,缓冲有延迟，视频已经播放了，但是加载指示器还会转动一会。
JPVideoPlayer.m -> startCheckBufferingTimer方法中的 currentTime > (self.playerModel.lastTime + 0.3) 这个判断是什么用处？ 我把0.3去掉貌似也不行，能请教一下这个如何修改么？
